libs/base/Control/IOExcept.idr

```diff
@@ -2,34 +2,25 @@ module Control.IOExcept
 
 -- An IO monad with exception handling
 
-data IOExcept : Type -> Type -> Type where
-     IOM : IO (Either err a) -> IOExcept err a
+record IOExcept err a where
+     constructor IOM
+     io : IO (Either err a)
```

Comment:
I'm not a fan of this projection name. Other, similar types use something like `runIOExcept`.